app/controllers/api/newsfeeds_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Api::NewsfeedsController < ApplicationController
+  def index
+    render json: Newsfeed.all.to_json
+  end
+
+  def show
+    newsfeed = Newsfeed.find(params[:id].to_i)
+    render json: newsfeed.to_json
+  rescue ActiveRecord::RecordNotFound
+    render json: nil, status: :not_found
```

Comment:
it is, but we have a test for an empty json body so I left it

Reply:
otherwise it tries to render a template by default and doesn't find it. I tried `head` too but it wouldn't let me do `head status: :not_found`. But I might just not know the magic words.

Reply:
`head :no_found` were the magic words

Reply:
rebased back into d5fd6fdea8e7158dc0ec7566b5e2d0c7ba1a9bd2